cajparser.py

```diff
@@ -322,6 +332,10 @@ def _convert_hn(self, dest):
             page_data = HNParsePage(output, page_style)
 
             if (images_per_page > 1):
+                if images_per_page > len(page_data.figures) or squared:
+                    from math import sqrt
+                    images_per_page = int(sqrt(images_per_page))
```

Comment:
There is no need to import sqrt from math : it is much better and much faster to check x * x = y, instead of x = int(sqrt(y)). 

Actually in most cases image_per_page is correct, and page_data.fugures is buggy - except when the sqrt is exact.

Reply:
So the preferred way is keep images_per_page, unless they are exact squares, in which case you overwrite images_per_page